### PR TITLE
Add a "link" tool

### LINKED ISSUE #285: How to test PRs?
If someone submit a PR with a patch for a bug, how can that be tested before the PR is merged?

### PR DESCRIPTION
**Pull request by jderusse**
Fixes #285

### CODE REVIEW

File: link [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+$packages = array();
+
+$filesystem = new Filesystem();
+$braces = array('.', 'Integration/*', 'Service');
+$directories = array_merge(...array_values(array_map(function ($part) {
+    return glob(__DIR__.'/src/'.$part.'/*', GLOB_ONLYDIR | GLOB_NOSORT);
+}, $braces)));
+
+$directories[] = __DIR__.'/src/Symfony/Contracts';
```
Comment (Nyholm) [inline link:38]: Is this needed?

File: link [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    if (!$copy && is_link($dir)) {
+        echo "\"$package\" is already a symlink, skipping.".PHP_EOL;
+        continue;
+    }
+
+    if (!isset($packages[$package])) {
+        continue;
+    }
+
+    $sfDir = ('\\' === DIRECTORY_SEPARATOR || $copy) ? $packages[$package] : $filesystem->makePathRelative($packages[$package], dirname(realpath($dir)));
```
Comment (Nyholm) [inline link:57]: `$sfDir`? Maybe rename it =)

File: link (hunk 1 above)
Comment (jderusse) [reply link]: good catch

File: link (hunk 2 above)
Comment (jderusse) [reply link]: thanks

File: link [hunk 3]
```diff
@@ -0,0 +1,65 @@
+#!/usr/bin/env php
+<?php
+
+require __DIR__.'/vendor/symfony/filesystem/Exception/ExceptionInterface.php';
+require __DIR__.'/vendor/symfony/filesystem/Exception/IOExceptionInterface.php';
+require __DIR__.'/vendor/symfony/filesystem/Exception/IOException.php';
+require __DIR__.'/vendor/symfony/filesystem/Filesystem.php';
```
Comment (stof) [inline link:7]: why explicit requires rather than requiring the autoloader ?

Comment (jderusse) [reply link:7]: Dummy copy/paste from Symfony's implementation

Comment (stof) [reply link:7]: The difference is that the Symfony script is usable without running `composer install`, as the require calls are in the package itself, not in `vendor`. But here, you rely on `composer install` having run already anyway (and so the autoloader will be there too).

Comment (jderusse) [reply link:7]: Ah ah.. That's so obvious.. Feel Dumb now..

Fixed in #498

**Pull request merged**